src/migrate/azext_migrate/_params.py

```diff
@@ -0,0 +1,148 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azext_migrate.actions import (
+    AddMachines
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('migrate location check-name-availability') as c:
+        c.argument('location_name', id_part=None, help='The desired region for the name check.')
+        c.argument('name', id_part=None, help='The name to check for availability')
+
+    with self.argument_context('migrate assessment-options show') as c:
+        c.argument('location_name', id_part=None, help='Azure region in which the project is created.')
+
+    with self.argument_context('migrate projects list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('migrate projects show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('project_name', id_part=None, help='Name of the Azure Migrate project.')
+
+    with self.argument_context('migrate projects create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('project_name', id_part=None, help='Name of the Azure Migrate project.')
+        c.argument('e_tag', id_part=None, help='For optimistic concurrency control.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('customer_workspace_id', id_part=None, help='ARM ID of the Service Map workspace created by user.')
+        c.argument('customer_workspace_location', id_part=None, help='Location of the Service Map workspace created by user.')
+        c.argument('provisioning_state', arg_type=get_enum_type(['Accepted', 'Creating', 'Deleting', 'Failed', 'Moving', 'Succeeded']), id_part=None, help='Provisioning state of the project.')

```

Comment:
didn't get it. provisioning_state is a readyonly property right? why it's a argument of create.

Reply:
This should have been removed in the latest code